common/types/head.go

```diff
@@ -17,10 +17,19 @@ type Head[BLOCK_HASH Hashable] interface {
 	// Hash is the head's block hash
 	BlockHash() BLOCK_HASH
 
+	// ParentHash is the head's parent block hash
+	GetParentHash() BLOCK_HASH
+
 	// Parent is the head's parent block
 	GetParent() Head[BLOCK_HASH]
 
 	// HashAtHeight returns the hash of the block at the given height, if it is in the chain.
 	// If not in chain, returns the zero hash
 	HashAtHeight(blockNum int64) BLOCK_HASH
 }
+
+type HeadTrackerHead[H Head[BLOCK_HASH], BLOCK_HASH Hashable] interface {
```

Comment:
I was going to suggest a pattern like this to avoid the `GetParent().(*Head)` casts at the bottom. What about moving all of the self-referential `Head[BLOCK_HASH]` methods from `Head` to this interface?

Reply:
With those methods moved, would it be accurate to call this a `HeadChain` or something like that?
```suggestion
type HeadChain[H Head[BLOCK_HASH], BLOCK_HASH Hashable] interface {
```

Reply:
I think that calling it `HeadChain` would be better since Headtracker uses the other self referential method. Other non headtracker services also makes use of `HeadChain` like eth confirmer.

Reply:
Fixed the issue of typecasting by exposing a wrapper method that allows concrete type to call GetHead() *Head or GetHead() Head